core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala

```diff
@@ -163,6 +163,14 @@ abstract class QuorumTestHarness extends Logging {
   // That way you control the initialization order.
   @BeforeEach
   def setUp(testInfo: TestInfo): Unit = {
+    Exit.setExitProcedure((code, message) => {
+      error(s"exit(${code}, ${message}) called!")
+      tearDown()
+    })
+    Exit.setHaltProcedure((code, message) => {
+      error(s"halt(${code}, ${message}) called!")
+      tearDown()
```

Comment:
I think this implementation for `setExitProcedure` and `setHaltProcedure` would allow the test that called `Exit.exit` and `Exit.halt` to continue. This is probably not what the user intended when they call those methods. If you agree, how about throwing an unchecked exception. Maybe `AssertionError`